GliaWidgets/SecureConversations/SecureConversations.Coordinator.swift

```diff
@@ -37,7 +37,13 @@ extension SecureConversations {
         }
 
         private func makeWelcomeViewModel() -> SecureConversations.WelcomeViewModel {
-            SecureConversations.WelcomeViewModel(
+            let uploadFile: () -> FileUpload.Environment.UploadFile = { [environment] in
+                return environment.getCurrentEngagement() == nil ?
```

Comment:
Here the decision to determine which endpoint to use it placed within coordinator. I would recommend to add unit test for this logic. Aslo I wonder if this logic can be moved to view model? View models seem to be common for testing.

Reply:
I have reworked the whole PR so the decision is made on the view model. Also tests were added.